hr_switzerland/models/res_users.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Stephane Eicher <seicher@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+import logging
+
+from odoo import models, api, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class ResUser(models.Model):
+    _inherit = 'res.users'
+
+    @api.model
+    def asterisk_connect(self, log=True):
+        for ast_user in self:
```

Comment:
Maybe filter only users that have `resource` field? `for ast_user in self.filtered('resource'):`